invenio_vocabularies/datastreams/writers.py

```diff
@@ -53,12 +54,19 @@ def __init__(self, service_or_name, identity, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
 
-    def write(self, stream_entry, *args, **kwargs):
-        """Writes the input entry using a given service."""
+    def write(self, stream_entry, force=False, *args, **kwargs):
+        """Writes the input entry using a given service.
+
+        :param force: if True update if exists, otherwise fail.
+        """
         try:
             result = self._service.create(self._identity, stream_entry.entry)
         except ValidationError as err:
             raise WriterError([{"ValidationError": err.messages}])
+        except PIDAlreadyExists as err:
```

Comment:
Would it be more desirable to do a `read()` instead of letting it raise a PK violation? I know in python we should "except" by default, but in this case we are generating an error at DB level.

The `read` is required anyway if `force == True` since we will require it to add changes to the record body.